java/custom/com/nokia/mid/ui/gestures/GestureRegistrationManager.java

```diff
@@ -13,41 +15,77 @@ public ZoneRegistration(Object container, GestureInteractiveZone zone) {
     }
 }
 
-public class GestureRegistrationManager {
+public class GestureRegistrationManager implements EventListener {
     public static final int MAX_SUPPORTED_GESTURE_INTERACTIVE_ZONES = 99;
 
     private static Vector zoneRegistrations = new Vector();
     private static Hashtable listenerRegistrations = new Hashtable();
 
-    public static void setListener(Object container, GestureListener listener) throws IllegalArgumentException {
-        if (listener != null) {
-            listenerRegistrations.put(container, listener);
-        } else {
-            listenerRegistrations.remove(container);
+    static {
+        EventQueue eventQueue = EventQueue.getEventQueue();
+        eventQueue.registerEventListener(EventTypes.GESTURE_EVENT, GestureRegistrationManager.getRegistrationManagerInstance());
+    }
+
+    private static GestureRegistrationManager gestureRegistrationManager = null;
+    private static GestureRegistrationManager getRegistrationManagerInstance() {
+        if (gestureRegistrationManager == null) {
+            gestureRegistrationManager = new GestureRegistrationManager();
```

Comment:
The getRegistrationManagerInstance method is only called by the static initializer (that is thread-safe and is called just once on class initialization).